### PR TITLE
IBMCloud: Using hive and hiveutil docs

### PR DESCRIPTION
**Pull request by abutcher**
[HIVE-1620](https://issues.redhat.com/browse/HIVE-1620)

### CODE REVIEW

File: docs/hiveutil.md [hunk 1]
```diff
@@ -119,6 +119,14 @@ clouds:
 bin/hiveutil create-cluster --cloud=openstack --openstack-api-floating-ip=192.168.1.2 --openstack-cloud=mycloud mycluster
 ```
 
+#### Create Cluster on IBM Cloud
+
+The IBM Cloud API key will be read from an `IC_API_KEY` environment variable. An [IBM Cloud credential manifests](./using-hive.md#ibm-cloud-credential-manifests) directory containing cloud credential secrets for OpenShift components must be provided.
+
+```bash
+bin/hiveutil create-cluster --cloud=ibmcloud --region="us-south" --base-domain=ibm.hive.openshift.com --manifests=~/manifests/ --credentials-mode-manual mycluster
```
Comment (2uasimojo) [inline docs/hiveutil.md:127]: Does hiveutil understand `~`? ISTR it doesn't.

File: docs/using-hive.md [hunk 2]
```diff
@@ -551,6 +615,13 @@ gcp:
 
 WARNING: Due to some naming restrictions on various components in GCP, Hive will restrict you to a max of 35 MachinePools (including the original worker pool created by default). We are left with only a single character to differentiate the machines and nodes from a pool, and 'm' is already reserved for the master hosts, leaving us with a-z (minus m) and 0-9 for a total of 35. Hive will automatically create a MachinePoolNameLease for GCP MachinePools to grab one of the available characters until none are left, at which point your MachinePool will not be provisioned.
 
+For IBM Cloud, replace the contents of `spec.platform` with:
+
+```yaml
+ibmcloud:
+  type: bx2-4x16
```
Comment (2uasimojo) [inline docs/using-hive.md:622]: ditto

**Review by 2uasimojo — COMMENTED** [review]
Looks reasonable to me, though of course I'm mostly having to take your word for it :)

Couple questions inline.

File: docs/hiveutil.md (hunk 1 above)
Comment (abutcher) [reply docs/hiveutil.md:127]: It does not. Will fix.

File: docs/using-hive.md [hunk 3]
```diff
@@ -350,6 +376,27 @@ and replace the contents of `platform` with:
     region: us-east1
 ```
 
+For IBM Cloud, replace the contents of `compute.platform` and `controlPlane.platform` with:
+
+```yaml
+  ibmcloud:
+    type: bx2-4x16
```
[the remark this answers is not in the document]
Comment (2uasimojo) [reply docs/using-hive.md:383]: Okay, so it's an instance type. Can that be explained on L379 (and similarly below)?

File: docs/using-hive.md [hunk 4]
```diff
@@ -350,6 +376,27 @@ and replace the contents of `platform` with:
     region: us-east1
 ```
 
+For IBM Cloud, replace the contents of `compute.platform` and `controlPlane.platform` with:
```
Comment (2uasimojo) [inline docs/using-hive.md:379]:
```suggestion
For IBM Cloud, replace the contents of `compute.platform` and `controlPlane.platform` with your desired instance type:
```

Comment (abutcher) [reply docs/using-hive.md]: I tacked this onto the end:

Note that `type` is any valid IBM Cloud instance type. `type` may be omitted to use OpenShift installation defaults.

**Pull request merged**